Pull Request Group 1

tklaass3 opened the pull request:
.

jjGG commented:
Hello Tim and Group1,

I was wondering what your idea was in the "cor_matrix_melted"?
-> you do get something but it looks like the correlation values are much closer to each other than before?

Melt is a function that is usally used to reshape data. e.g. From longformat to wide 

What did you think you were doing here?

cu - jonas

```
library(RColorBrewer)
# Reshape the correlation matrix for ggplot
cor_matrix_melted <- melt(cor_matrix)

# Create a Pearson correlation heatmap
ggplot(cor_matrix_melted, aes(Var1, Var2, fill = value)) +
  geom_tile() +  # Use tiles to visualize correlation
  scale_fill_gradient2(low = "blue", high = "yellow", mid = "green", midpoint = 0, 
                       limits = c(-1, 1), name = "Correlation") +  # Color scale
  theme_minimal() +
  labs(title = "Pearson Correlation Heatmap of Protein Expression Data",
       x = "Samples", y = "Samples") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1),  # Rotate x-axis labels
        axis.text.y = element_text(angle = 0, hjust = 1)) +  # Keep y-axis labels horizontal
  coord_fixed()  # Fix aspect ratio to make it square


```

jjGG commented:
I am there fore not yet accepting this PR. 
Please either remove it (you do have the corr-matrix already visualized as heatmap.. 
or
explain what we see in this last plot?

tklaass3 commented:
Hello Jonas,

That part of code was indeed a bit surprising since it wasn't part of the original packages provided in the tutorial. Since I was running out of time I asked genAI if it could help me create this Pearson cor heatmap. To do this with ggplot it used this melt function from the Rcolorbrewer function to reshape the data to the long format so ggplot could make a heatmap out if it. 
I took some extra time and removed this code and changed it to code using the pheatmap package. It should now create a plot more in line with what you would expect (hopefully). I don't know if this update is already in this pull request or if I should request again?
Thanks in advance!

Best regards,

Tim

jjGG commented:
Hello Tim, 

Perfect - thanks a lot. 

But: 
1) Rcolorbrewer is only for adjusting colors
2) reshape2::melt function is to "untidy" your data from longformat to wide
https://favtutor.com/blogs/melt-function-in-r

Your PR is accepted now. Thanks a lot for your extra effort.
j
